docs/rules/jsx-sort-props.md

```diff
@@ -75,6 +79,34 @@ When `true`, alphabetical order is not enforced:
 <Hello tel={5555555} name="John" />
 ```
 
+### `reservedFirst`
+
+This can be a boolean or an object option.
+
+When `reservedFirst` is defined, React reserved props (default: `children`, `dangerouslySetInnerHTML`, `htmlFor`, `key`, `ref`, `style`) must be listed before all other props, but still respecting the alphabetical order:
+
+```jsx
+<Hello key={0} ref="John" name="John" />
+```
+
+If given as an object, the option accepts two optional sub-options:
+
+#### `reservedList`
+
+Use this to optionally override the default list of reserved props.
```

Comment:
I'm open to that; I was hoping to let people know what the rule is doing without making them read all the options.